components/cloud/aws/src/s3.rs

```diff
@@ -626,6 +640,68 @@ impl BlobStorage for S3Storage {
     }
 }
 
+struct S3Walker<'cli, 'arg> {
```

Comment:
Maybe `S3PrefixIter`?

Reply:
hmm, can the prefix be nil? Not sure about the detail implementation but feels like it can be, in that case it will just scan the entire space without prefix. I guess maybe  `PrefixIter` is not generic enough since the `ListObjectV2` can totally work without any prefix. Let me know your thoughts.